src/main/kotlin/interpreter/KythonInterpreter.kt

```diff
@@ -148,26 +134,28 @@ object KythonInterpreter {
      */
     fun kickoffThread(frame: StackFrame, child: Boolean = true) {
         try {
-            this.runPythonThread(frame)
+            runPythonThread(frame)
         } catch (e: Throwable) {  // blah blah, bad practice, who cares
 
             System.err.println("Fatal interpreter error!")
             e.printStackTrace(System.err)
             System.err.println("\nKython stack (most recent frame first):")
 
             val stacks = StackFrame.flatten(frame).reversed()
-            for (frame in stacks) {
-                val info = frame.getStackFrameInfo()
-                System.err.println("   " + info.getTracebackString())
-                if (info.hasDisassembly) {
-                    val disassembly = info.getDisassembly()
-                    System.err.println("Disassembly:\n$disassembly")
-                }
-                if (info.hasStack) {
-                    val stack = info.getStack()
-                    System.err.println("Function stack, size: ${stack.size}")
-                    for ((idx, pyo) in stack.iterator().withIndex()) {
-                        System.err.println("    $idx: $pyo")
+            stacks.forEach {
+                with(it.getStackFrameInfo()) {
+                    System.err.println("   " + getTracebackString())
+                    if (hasDisassembly) {
+                        System.err.println("Disassembly:\n${getDisassembly()}")
+                    }
+
+                    if (hasStack) {
+                        getStack().run {
+                            System.err.println("Function stack, size: $size")
+                            forEachIndexed { index, pyObject ->
+                                System.err.println("    $index: $pyObject")
```

Comment:
This is too nested (and, by extension, indented). I'd prefer the old way or even removing that `.run` and just use a temporary variable instead.

Reply:
Sure. I can remove the `.run`.